crates/iota-core/src/consensus_adapter.rs

```diff
@@ -182,6 +182,87 @@ impl ConsensusAdapterMetrics {
+        registry
+            .unregister(Box::new(self.sequencing_certificate_failures.clone()))
+            .expect("Failed to unregister sequencing_certificate_failures");
+        registry
+            .unregister(Box::new(self.sequencing_certificate_inflight.clone()))
+            .expect("Failed to unregister sequencing_certificate_inflight");
+        // Use custom method to unregister self.sequencing_acknowledge_latency
+        unregister_histogram_vec(
+            "sequencing_acknowledge_latency",
+            "The latency for acknowledgement from sequencing engine. The overall sequencing latency is measured by the sequencing_certificate_latency metric",
+            &["retry", "tx_type"],
+            registry,
+        );
+        registry
+            .unregister(Box::new(self.sequencing_certificate_latency.clone()))
+            .expect("Failed to unregister sequencing_certificate_latency");
+        registry
+            .unregister(Box::new(
+                self.sequencing_certificate_authority_position.clone(),
+            ))
+            .expect("Failed to unregister sequencing_certificate_authority_position");
+        registry
+            .unregister(Box::new(
+                self.sequencing_certificate_positions_moved.clone(),
+            ))
+            .expect("Failed to unregister sequencing_certificate_positions_moved");
+        registry
+            .unregister(Box::new(
+                self.sequencing_certificate_preceding_disconnected.clone(),
+            ))
+            .expect("Failed to unregister sequencing_certificate_preceding_disconnected");
+        registry
+            .unregister(Box::new(self.sequencing_certificate_processed.clone()))
+            .expect("Failed to unregister sequencing_certificate_processed");
+        registry
+            .unregister(Box::new(self.sequencing_in_flight_semaphore_wait.clone()))
+            .expect("Failed to unregister sequencing_in_flight_semaphore_wait");
+        registry
+            .unregister(Box::new(self.sequencing_in_flight_submissions.clone()))
+            .expect("Failed to unregister sequencing_in_flight_submissions");
+        registry
+            .unregister(Box::new(self.sequencing_estimated_latency.clone()))
+            .expect("Failed to unregister sequencing_estimated_latency");
+        registry
+            .unregister(Box::new(self.sequencing_resubmission_interval_ms.clone()))
+            .expect("Failed to unregister sequencing_resubmission_interval_ms");
+    }
+}
+
+// HistogramVec from iota-metrics uses asynchronous model to report metrics
+// which makes it difficult to unregister counters in the usual manner. Here we
+// re-create counters so that their `desc()`s match the ones created by
+// HistogramVec and remove them from the registry. Counters can be safely
+// unregistered even if they are still in use.
+fn unregister_histogram_vec(name: &str, desc: &str, labels: &[&str], registry: &Registry) {
```

Comment:
I wonder if that should rather be moved to `histogram.rs` in `iota-metrics`, below the logic where it's created/registered.